### PR TITLE
Made it so that search options query on data collection, similar docs…

### PR DESCRIPTION
**Pull request by hunterwilliams**
#161 
#136 

Made it so that similar, search options, and people creation properly utilize the data collection.

### CODE REVIEW

File: app/templates/ui/app/create/create.controller.js [hunk 1]
```diff
@@ -44,7 +44,8 @@
       mlRest.createDocument(ctrl.person, {
         format: 'json',
         directory: '/content/',
-        extension: '.json'
+        extension: '.json',
+        collection: ['data','data/person']
```
Comment (grtjn) [inline app/templates/ui/app/create/create.controller.js]: Was it data/person, or data/people?

**Comment by hunterwilliams** [comment]
@grtjn Nice catch. I changed it on my sample project and thought I committed that...

**Comment by joemfb** [comment]
Could the collection name be an optional parameter to `extsimilar`?

**Comment by hunterwilliams** [comment]
Now should work! @joemfb 

If you don't pass a collection it just searches over all documents

**Comment by grtjn** [comment]
You could also consider sending options name, pull up persisted options, blend in similar query, and do a search:search. But I could have a whack at that myself. Might wanna move it to an mlpm package as well..

**Comment by grtjn** [comment]
search-utils can be used to grab persisted options, see example here: https://github.com/grtjn/ml-decorators/blob/master/facet-values-decorator.xqy#L14

**Comment by hunterwilliams** [comment]
@grtjn my only concern / possible annoyance would be that you want to filter further than your general search options and just to change this you'd have to make another search options. I see the benefits but if a developer is going to really need custom search options for each type they're likely savvy enough to fix the directive for their particular use.

**Comment by grtjn** [comment]
Not necessarily. You could make it such that you can also send along a string query, or a structured query which is more powerful than just a collection param. Then again, this really sounds like rebuilding /v1/search, which already takes all these params, and allows sending along override query options etc. Can't we just send a cts:similar-query to that somehow? Maybe I need to have a closer look at this..

**Comment by hunterwilliams** [comment]
I mean you could yank the collections from the current document and use that in the search or it could possibly grab the all search options and do the similar query with those.

**Comment by grtjn** [comment]
Interesting thought pulling collections from the doc in question itself. :)

/v1/search allows return similar, by the way. Bit awkward with binaries, but could be a sensible alternative..

**Comment by hunterwilliams** [comment]
Going to just make default to the document's collections by default and remove the stuff from the directive. Going with the keep it simple approach after talking with @grtjn

**Comment by grtjn** [comment]
Yes, we can work on a more advanced extsimilar as a separate component if we think that is useful. A version that would go much more towards /v1/search would be interesting. Should also handle binaries better.

Looking forward to your updates! ;)

**Comment by hunterwilliams** [comment]
Ready to be verified and merged. May need to squash down commits

**Comment by grtjn** [comment]
I'm happy, but let's hold off a little if anyone else has comments..

**Comment by grtjn** [comment]
The details page has been touched in a different PR. I'm afraid that caused a merge conflict. Can you undo that detail page change, and then rebase?

**Comment by hunterwilliams** [comment]
Rebased and updated

**Comment by grtjn** [comment]
Perfect, runs smoothly for me..

**Pull request merged**